src/java.base/share/classes/sun/security/rsa/RSASignature.java

```diff
@@ -190,7 +190,10 @@ protected byte[] engineSign() throws SignatureException {
         try {
             byte[] encoded = RSAUtil.encodeSignature(digestOID, digest);
             byte[] padded = padding.pad(encoded);
-            return RSACore.rsa(padded, privateKey, true);
+            if (padded != null) {
+                return RSACore.rsa(padded, privateKey, true);
+            }
+            throw new SignatureException("Could not sign data");
```

Comment:
It may be clearer if the throw line is moved to the end of the method.  Otherwise, I have to check if SignatureException is a sub-class of GeneralSecurityException.

Reply:
Yes, I think so too.